### DISCUSSION TITLE
RPI2040 + PlatformIo

### DISCUSSION THREAD

**Discussion by Jaj0** [tttapa/Control-Surface#878] [Q&A]
Hello,
has anyone managed to sucesfully use this library with PlatformIO and RPI Pico?

In library manager im able to see version 1.2.0 and simple code does not compile, with issues being MIDIUSB not recognizing RPI2040 as device with USB. So i've downloaded zip with this repo and provided it directly (which gives me version 2.0.0). But error still persists.
Simple code i try to compile:
```
#include <Arduino.h>
#include <Control_Surface.h>

USBMIDI_Interface midi;

void setup() {
  Control_Surface.begin();
}

void loop() {
  Control_Surface.loop();
}
```

My platformio.ini:
```
[env:pico]
platform = raspberrypi
board = pico
framework = arduino
lib_deps = E:\Downloads\Control-Surface-main.zip
```
Beggining of error:

> In file included from .pio\libdeps\pico\MIDIUSB\src\MIDIUSB.cpp:17:
.pio\libdeps\pico\MIDIUSB\src\MIDIUSB.h:18:2: error: #error MIDIUSB can only be used with an USB MCU.
   18 | #error MIDIUSB can only be used with an USB MCU.
      |  ^~~~~
In file included from .pio\libdeps\pico\MIDIUSB\src\MIDIUSB.cpp:17:
.pio\libdeps\pico\MIDIUSB\src\MIDIUSB.h:78:2: error: #error "Unsupported architecture"
   78 | #error "Unsupported architecture"
      |  ^~~~~
.pio\libdeps\pico\MIDIUSB\src\MIDIUSB.h:203: warning: "WEAK" redefined
  203 | #define WEAK __attribute__ ((weak))
      |
In file included from C:\Users\mateu\.platformio\packages\framework-arduino-mbed\cores\arduino/mbed/platform/include/platform/mbed_assert.h:21,
                 from C:\Users\mateu\.platformio\packages\framework-arduino-mbed\cores\arduino/mbed/platform/include/platform/Callback.h:24,
                 from C:\Users\mateu\.platformio\packages\framework-arduino-mbed\cores\arduino/mbed/platform/include/platform/FileHandle.h:23,
                 from C:\Users\mateu\.platformio\packages\framework-arduino-mbed\cores\arduino/macros.h:41,
                 from C:\Users\mateu\.platformio\packages\framework-arduino-mbed\variants\RASPBERRY_PI_PICO/pins_arduino.h:2,
                 from C:\Users\mateu\.platformio\packages\framework-arduino-mbed\cores\arduino/Arduino.h:76,
                 from .pio\libdeps\pico\MIDIUSB\src\MIDIUSB.h:11,
                 from .pio\libdeps\pico\MIDIUSB\src\MIDIUSB.cpp:17:

Is there some additional steps which need to be done in order to make this lib working with PlatformIO? Maybe some other arduino core? I don't know how to change them.

When i install this library from zip into arduino 1.8.16 it compiles correctly.

My PlatformIO is `Core 6.1.5 Home 3.4.3`
Used Packages:
```
 - framework-arduino-mbed @ 3.1.1
 - tool-rp2040tools @ 1.0.2
 - toolchain-gccarmnoneeabi @ 1.90201.191206 (9.2.1)
 ```

**Comment by tttapa** [comment]
This is an issue with the PlatformIO library dependency resolver. For future reference, see the workaround in #309.
